site-src/geps/gep-726.md

```diff
@@ -0,0 +1,266 @@
+# GEP-726: Add Path Redirects and Rewrites
+
+* Issue: [#726](https://github.com/kubernetes-sigs/gateway-api/issues/726)
+* Status: Implementable
+
+## TLDR
+
+This GEP proposes adding support for path redirects and rewrites in addition to
+host rewrites. This would augment the existing host redirection capabilities.
+
+## Goals
+
+* Implement path redirects.
+* Implement the most portable and simple forms of path rewrites.
+* Describe how more advanced rewrite and redirect and redirect capabilities
+  could be added in the future.
+
+## API
+
+Although many implementations support very advanced rewrite and redirect
+capabilities, the following are the most [portable](#portability) concepts that
+are not already supported by the Gateway API:
+
+* Path redirects
+* Path prefix redirects
+* Path prefix rewrites
+* Host rewrites
+
+Although regular expression based redirects and rewrites are commonly supported,
+there is significantly more variation in both if and how they are implemented.
+Given the wide support for this concept, it is important to design the API in a
+way that would make it easy to add this capability in the future.
+
+### Redirects
+
+The existing `RequestRedirect` filter can be expanded to support path redirects:
+
+```yaml
+kind: HTTPRoute
+apiVersion: gateway.networking.k8s.io/v1alpha2
+metadata:
+  name: http-filter-1
+spec:
+  rules:
+    - matches:
+      - path:
+          type: Prefix
+          value: /
+      filters:
+      - type: RequestRedirect
+        requestRedirect:
+          hostname: foo.com
+          path:
+            type: Prefix
+            value: /bar
```

Comment:
Suggest that we use something like `/app` as the matching prefix. Then we can give a clear example of exactly what redirect location header is generated.

Reply:
Updated to use a clearer example for both this and rewrites.